### PR TITLE
feat: explicitly allow paradigm sizes

### LINKED ISSUE #853: Feature: pane-based paradigm layout sizes should have configurable order
The order of the pane-based paradigm layouts should be configurable. Perhaps a setting in `settings.py`:

```python
MORPHODICT_PARADIGM_SIZES = [
    "basic",
    "full",
]
```

There should be a check that all `*.tsv` layout files that specify a size exist in this `setting`, otherwise, some sort of warning or error is reported (maybe a [system check](https://docs.djangoproject.com/en/3.2/topics/checks/)?).

### PR DESCRIPTION
**Pull request by eddieantonio**
# What's in this PR?

 - **added**: `MORPHODICT_PARADIGM_SIZES` setting for `crkeng`. A list where you list the paradigm sizes!
 - **added**: `ParadigmManagerWithExplicitSizes` is a `ParadigmManager` that maintains a sorted list of paradigm sizes
   - `ParadigmManagerWithExplicitSizes.sizes_of(paradigm_name)` always returns a sorted list
   - `ParadigmManagerWithExplicitSizes.all_sizes_fully_specified()` can help check for misconfiguration errors
 - **added**: an integration test for `crkeng` that asserts that the paradigm sizes are ordered, and that all sizes are fully specified

Fixes #853

### CODE REVIEW

File: src/CreeDictionary/CreeDictionary/paradigm/manager.py [hunk 1]
```diff
@@ -130,6 +130,59 @@ def _inflect(self, layout: ParadigmLayout, lemma: str) -> Paradigm:
         return layout.fill(template2forms)
 
 
+class ParadigmManagerWithExplicitSizes(ParadigmManager):
```
Comment (eddieantonio) [inline src/CreeDictionary/CreeDictionary/paradigm/manager.py:133]: Note: this class could be a [decorator](https://python-patterns.guide/gang-of-four/decorator-pattern/) (gang of four design pattern), but that would involve an Extract Interface refactoring to make Python's static type system happy. I decided that's just too much, after I've already done _too much_ in this PR!

File: src/crkeng/site/settings.py [hunk 2]
```diff
@@ -66,3 +66,14 @@
...
+# The order in which paradigm sizes will be presented to the user.
+# The first size in this list is the "default".
+# Make sure to exhaustively specify all size options available!
+MORPHODICT_PARADIGM_SIZES = [
+    # The most user-friendly size should be first:
+    "basic",
+    # Then, a more complete paradigm layout:
+    "full",
+    # Variants for linguists go here:
+]
```
Comment (eddieantonio) [inline src/crkeng/site/settings.py:79]: @aarppe this is where you would add "algonquianist" paradigm size option (#860).

File: src/crkeng/app/integration_tests/test_paradigm_manager.py [hunk 3]
```diff
@@ -8,7 +8,17 @@
...
+    ParadigmManager,
+    ParadigmManagerWithExplicitSizes,
+)
+
+
+def test_paradigm_sizes_are_ordered(paradigm_manager):
+    assert isinstance(paradigm_manager, ParadigmManagerWithExplicitSizes)
+    assert (
+        paradigm_manager.all_sizes_fully_specified()
+    ), "please check MORPHODICT_PARADIGM_SIZES or the names of the .tsv files"
```
Comment (eddieantonio) [inline src/crkeng/app/integration_tests/test_paradigm_manager.py:21]: This is intended to fail the CI if files in `res/layouts` are out of place, or if `settings.PARADIGM_SIZES` is misconfigured.

**Review by andrewdotn — APPROVED** [review]

File: src/CreeDictionary/CreeDictionary/paradigm/test_manager.py [hunk 4]
```diff
@@ -91,3 +153,17 @@ def bulk_lookup(self, analyses: Iterable[str]) -> dict[str, set[str]]:
...
+def distinct_permutation(sequence):
+    """
+    Returns a permutation of the given sequence that is guaranteed to not be the same
+    order as the given input.
+    """
+    assert len(sequence) > 1, "there is no distinct permutation for this input"
+    original = list(sequence)
+    result = original.copy()
+
+    while result != original:
```
Comment (nienna73) [inline src/CreeDictionary/CreeDictionary/paradigm/test_manager.py:167]: Doesn't this guarantee that `result == original`?
```suggestion
    while result == original:
```

**Review by nienna73 — APPROVED** [review]
I have one question, otherwise looks good!

File: src/CreeDictionary/CreeDictionary/paradigm/test_manager.py (hunk 4 above)
Comment (eddieantonio) [reply src/CreeDictionary/CreeDictionary/paradigm/test_manager.py:167]: yes 🤯  why... why does this code work...

**Pull request merged**